compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/evaluate/IrConstTransformer.kt

```diff
@@ -60,18 +61,33 @@ class IrConstTransformer(irBuiltIns: IrBuiltIns) : IrElementTransformerVoid() {
             for (i in 0 until annotation.valueArgumentsCount) {
                 val arg = annotation.getValueArgument(i) ?: continue
                 when (arg) {
-                    is IrVararg -> arg.transformVarArg()
+                    is IrVararg -> annotation.putValueArgument(i, arg.transformVarArg())
                     else -> annotation.putValueArgument(i, arg.transformSingleArg(annotation.symbol.owner.valueParameters[i].type))
                 }
             }
         }
     }
 
-    private fun IrVararg.transformVarArg() {
-        for (i in this.elements.indices) {
-            val irVarargElement = this.elements[i] as? IrExpression ?: continue
-            this.putElement(i, irVarargElement.transformSingleArg(this.varargElementType))
+    private fun IrVararg.transformVarArg(): IrVararg {
+        if (elements.isEmpty()) return this
+        val newIrVararg = IrVarargImpl(this.startOffset, this.endOffset, this.type, this.varargElementType)
+        for (element in this.elements) {
+            when (element) {
+                is IrExpression -> newIrVararg.addElement(element.transformSingleArg(this.varargElementType))
+                is IrSpreadElement -> {
+                    when (val expression = element.expression) {
+                        is IrVararg -> {
```

Comment:
I would prefer here single line representation
`is IrVararg -> expression.transformVarArg().elements.forEach { newIrVararg.addElement(it) }`

Reply:
Thank you for the suggestion! Done.